### PR TITLE
Improvement: Update for Moodle 4.4 compatibility and improved MathJax support 1658984705

### PR DESCRIPTION
**Pull request by prasanna-lmsace**
- Improved MathJax rendering for better equation display in Moodle 4.4.
- Updated codebase to ensure full compatibility with the latest Moodle version.

### CODE REVIEW

File: amd/src/initmathjax.js [hunk 1]
```diff
@@ -1,22 +1,61 @@
...
-            // MathJax has not yet been loaded.
-          }
-        }, 100);
-      });
-    },
-  };
+
+            // Callaback to the observer.
+            function callback(mutationList, observer) {
+                // console.log('mutation', mutationList);
```
Comment (HAASLEWER) [inline amd/src/initmathjax.js:42]: @prasanna-lmsace I think you forgot to remove this.

File: amd/src/initmathjax.js [hunk 2]
```diff
@@ -1,22 +1,61 @@
...
-        // Initialise MathJax typesetting
-        var nodes = Y.all(".latex-math");
-        Y.fire(M.core.event.FILTER_CONTENT_UPDATED, { nodes: nodes });
-
-        var waitForMathJax = setInterval(function () {
-          try {
-            if (MathJax) {
-              MathJax.Hub.Queue(["Typeset", MathJax.Hub]);
-              clearInterval(waitForMathJax);
+define(["jquery", "core/ajax", 'core_filters/events'], function ($, Ajax, FilterEvents) {
```
Comment (HAASLEWER) [inline amd/src/initmathjax.js:1]: This JS is much better than the original well done and thank you!

File: amd/src/initmathjax.js [hunk 3]
```diff
@@ -1,22 +1,61 @@
...
+                        (e) => e.classList.add(SELECTORS.classes.filterMathjax)
+                    );
+
+                    // Notifiy the siyavula filtered content are updated.
+                    FilterEvents.notifyFilterContentUpdated(root);
+
+                    // Ensure MathJax is ready before triggering typeset
+                    if (window.MathJax) {
+                        if (window.MathJax.Hub.queue.pending) {
+                            window.MathJax.Hub.Queue(["Typeset", window.MathJax.Hub, root]);
```
Comment (HAASLEWER) [inline amd/src/initmathjax.js:34]: I noticed that MathJax takes slightly longer to typeset than it did for the previous version, is there a way we can optimise typesetting, please?

Comment (prasanna-lmsace) [reply amd/src/initmathjax.js]: Thank you for your feedback! We’ve identified the performance issue with MathJax typesetting and have implemented optimizations to improve its speed

File: amd/src/initmathjax.js (hunk 1 above)
Comment (prasanna-lmsace) [reply amd/src/initmathjax.js]: Removed the unused codes

File: version.php [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <?php
 
 defined('MOODLE_INTERNAL') || die();
-$plugin->version   = 2023121300;        // The current plugin version (Date: YYYYMMDDXX)
+$plugin->version   = 2023121301;        // The current plugin version (Date: YYYYMMDDXX)
 $plugin->requires  = 2019111809;        // Requires this Moodle version
```
[the remark this answers is not in the document]
Comment (prasanna-lmsace) [reply version.php:5]: Hi @HAASLEWER,
Yes, the recent changes have been implemented with backward compatibility

[the remark this answers is not in the document]
Comment (HAASLEWER) [reply version.php:5]: @prasanna-lmsace It's working perfectly, thanks!

**Review by HAASLEWER — APPROVED** [review]

**Pull request merged**